src/storage/v2/durability/snapshot.cpp

```diff
@@ -127,6 +139,230 @@ struct BatchInfo {
+      }
+      // TODO Regenerate batches (currently could be less then optimal)
+      edge_batch_infos.insert(edge_batch_infos.end(), res.batch_info.begin(), res.batch_info.end());
+      // Append the edge part to the snapshot
+      int edges_fd = open(res.snapshot_path.string().c_str(), O_RDONLY);
+      if (edges_fd == -1) {
+        throw RecoveryFailure("Couldn't open edge snapshot part!");
+      }
+      utils::OnScopeExit cleanup{[&] {
+        close(edges_fd);
+        std::error_code ec;
+        std::filesystem::remove(res.snapshot_path, ec);
+        if (ec) spdlog::warn("Couldn't remove temporary edge snapshot part {}: {}", res.snapshot_path, ec.message());
+      }};
+      // Use sendfile for efficient copying (zero-copy)
+      off_t offset = 0;
+      const auto size = res.snapshot_size - offset;
+      ssize_t bytes_sent = sendfile(snapshot.native_handle(), edges_fd, &offset, size);
+      if (bytes_sent == -1 || bytes_sent != size) {
+        throw RecoveryFailure("Couldn't copy edge part to snapshot!");
+      }
+      snapshot.SetPosition(current_offset + size);
+    }
+  }
+
+  i = 0;
+  offset_vertices = snapshot.GetPosition();  // Global vertex offset
+  for (auto &[res, promise] : vertex_res) {
+    promise.get_future().wait();  // Wait for incoming result
+    spdlog::trace("Handling snapshot vertex part {}, size {}, count {}...", i++, res.snapshot_size, res.count);
+    if (res.snapshot_size > 0) {
+      vertices_count += res.count;
+      used_ids.merge(std::move(res.used_ids));
+      const auto current_offset = snapshot.GetPosition();
+      // Update batch positions
+      for (auto &[offset, _] : res.batch_info) {
+        offset += current_offset;
+      }
+      // TODO Regenerate batches (currently could be less then optimal)
+      vertex_batch_infos.insert(vertex_batch_infos.end(), res.batch_info.begin(), res.batch_info.end());
+      // Append the vertex part to the snapshot
+      int vertex_fd = open(res.snapshot_path.string().c_str(), O_RDONLY);
+      if (vertex_fd == -1) {
+        throw RecoveryFailure("Couldn't open vertex snapshot part!");
+      }
+      utils::OnScopeExit close_vertex_fd([&] {
+        close(vertex_fd);
+        std::error_code ec;
+        std::filesystem::remove(res.snapshot_path, ec);
+        if (ec) spdlog::warn("Couldn't remove temporary vertex snapshot part {}: {}", res.snapshot_path, ec.message());
+      });
+      // Use sendfile for efficient copying (zero-copy)
+      off_t offset = 0;
+      const auto size = res.snapshot_size - offset;
+      ssize_t bytes_sent = sendfile(snapshot.native_handle(), vertex_fd, &offset, size);
+      if (bytes_sent == -1 || bytes_sent != size) {
+        throw RecoveryFailure("Couldn't append vertex part to snapshot!");
+      }
+      snapshot.SetPosition(current_offset + size);
+    }
+  }
+}
+
+// Return at least one batch (at least 2 elements: start and end gid)
+auto Batch(auto &&acc, const uint64_t step) {
```

Comment:
maybe rename step to items_per_batch, makes it more clear to me what's the purpose of it